prov/verbs/src/verbs_cq.c

```diff
@@ -227,7 +227,7 @@ int vrb_poll_cq(struct vrb_cq *cq, struct ibv_wc *wc)
 		ctx = (struct vrb_context *) (uintptr_t) wc->wr_id;
 		wc->wr_id = (uintptr_t) ctx->user_ctx;
 		if (wc->status != IBV_WC_SUCCESS && wc->status != IBV_WC_WR_FLUSH_ERR)
-			vrb_shutdown_qp_in_err(ctx->ep);
+			vrb_shutdown_ep(ctx->ep);
```

Comment:
I don't know what non-IBV_WC_SUCCESS would result in QP transitioning to IBV_QPS_ERR by the adapter.  That's the reason behind ibv_query_qp to make sure we are only closing on QP in IBV_QPS_ERR.  This would potentially close QP that's still in RTS with a non-fatal status.  Also, don't think status and behavior would be the same between HW vendors thus checking current qp state is the only safe way to do this.